bin/stt_test/stt_providers/ISTTBase.py

```diff
@@ -0,0 +1,69 @@
+import json
+import logging
+from typing import List, Literal, Protocol, TypedDict
+import speech_recognition as sr  # type: ignore
+import time
+from stt_test.utils.SentenceChecker import SentenceChecker
+from stt_test.utils.test_data import get_audio_with_transcription
+import inspect
+
+
+class STTResult(TypedDict):
+    stt_trans: str
+    trans: str
+    similarity: float
+    execution_time: float
+
+
+class ISSTBase(Protocol):
+
+    MODEL_SIZES: List[str]
+    _sc: SentenceChecker
+    _recognizer: sr.Recognizer
+
+    def _init(self):
```

Comment:
Constructor?

Reply:
No, it's just a private "init" method. We can change the name if you think so

Reply:
Yup, but can't we use a constructor instead?

Reply:
https://stackoverflow.com/questions/62768327/typing-protocol-class-init-method-not-called-during-explicit-subtype-const